modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_3_0/UpgradeDDMStructureLinkDLFileEntryType.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.internal.upgrade.v3_3_0;
+
+import com.liferay.document.library.kernel.model.DLFileEntryMetadata;
+import com.liferay.document.library.kernel.model.DLFileEntryType;
+import com.liferay.document.library.kernel.util.DLUtil;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortalUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Alicia Garcia
+ */
+public class UpgradeDDMStructureLinkDLFileEntryType extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_fixDataDefinitionIdFields();
+
+		_populateFields();
+	}
+
+	private void _fixDataDefinitionIdFields() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+			"select fileEntryTypeId, DLFileEntryType.companyId, " +
+			"dataDefinitionId from DLFileEntryType inner join DDMStructure " +
+			"ON dataDefinitionId = structureId where type_ = 0");
+
+			 PreparedStatement preparedStatement2 =
+				 AutoBatchPreparedStatementUtil.autoBatch(
+					 connection.prepareStatement(
+						 "update DLFileEntryType set dataDefinitionId = 0 " +
+						 "where fileEntryTypeId = ? "));
+
+			 PreparedStatement preparedStatement3 =
```

Comment:
This will fail if there's already a DDMStructureLink. Are we sure that cannot happen?

Also, there is an SF problem. There should be a newline after the autoBatch( call; the string indentations seems to be wrong too.

Reply:
> This will fail if there's already a DDMStructureLink. Are we sure that cannot happen?
Added a check if exist the same reg that is added.


> Also, there is an SF problem. There should be a newline after the autoBatch( call; the string indentations seems to be wrong too.
Seems that is the way the SF format it, I tried to make as you said and